brian2/stateupdaters/explicit.py

```diff
@@ -570,10 +570,10 @@ def __call__(self, eqs, variables=None, method_options=None):
         v = _v
         >>> print(rk4(eqs))
         __k_1_v = -dt*v/tau
-        __k_2_v = -dt*(0.5*__k_1_v + v)/tau
-        __k_3_v = -dt*(0.5*__k_2_v + v)/tau
+        __k_2_v = -dt*(__k_1_v/2 + v)/tau
+        __k_3_v = -dt*(__k_2_v/2 + v)/tau
         __k_4_v = -dt*(__k_3_v + v)/tau
-        _v = 0.166666666666667*__k_1_v + 0.333333333333333*__k_2_v + 0.333333333333333*__k_3_v + 0.166666666666667*__k_4_v + v
+        _v = __k_1_v/6 + __k_2_v/3 + __k_3_v/3 + __k_4_v/6 + v
```

Comment:
This I'm not sure about. If we know that ``x`` is a float, then we can safely convert ``x/2`` to ``0.5*x`` and this will be faster. Probably the compiler does it already if ``-funsafe-math-optimizations`` is on, but would be interesting to know if there are cases where the new version would cause it to run slower. What do you think?

Reply:
This is not really about optimized code here: when we define the RK4 state  updater, we write things like `k_3 = dt*f(x+k_2/2,t+dt/2)` – before, sympy converted this to floating point numbers and to multiplications along the way. I don't think this is for sympy to do where we are still at the level of mathematics. Optimizing the generated code is a different question, I don't think we currently turn divisions into multiplications but we could (even though I doubt it makes a measurable difference). We can run our benchmark suite after merging, but I'm not really worried about this. I'm more worried that results change slightly due to the numerical difference, this might be worth investigating further.

Reply:
Right, I agree that sympy shouldn't be doing that for us because it doesn't know that it might be relevant that it's an integer.